### PR TITLE
DOC and BUG: Bessel function docstring improvements, fix array_like, general formatting

### PR DESCRIPTION
**Pull request by endolith**
1. General formatting like spaces after commas, italicize variables, link function names
2. Added examples, parameters, returns for Bessel K functions
3. Bessel derivatives should handle array_like for function orders, but only accept true arrays, fixed that

### CODE REVIEW

**Comment by endolith** [comment]
Doctest fails? Should I say special.kvp instead of kvp?

**Comment by ev-br** [comment]
You need to import them, I think. The only implicit import is `import numpy as np`.

**Comment by pv** [comment]
You need to write the doctest so that it runs assuming only that
`import numpy as np` has been done.

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -1952,32 +1986,65 @@ def add_newdoc(place, name, doc):
...
+    out : ndarray
+        The results
+
+    Examples
+    --------
+    Plot the function of several orders for real input:
+
+    >>> import matplotlib.pyplot as plt
+    >>> for N in linspace(0, 2, 5):
+    >>>     plot(x, kv(N, x), label='$K_{{{}}}(x)$'.format(N))
```
Comment (pv) [inline scipy/special/add_newdocs.py]: plt.plot, and the continuation line should start with `...` instead of `>>>`

Comment (pv) [inline scipy/special/add_newdocs.py]: @ev-br: the doctest checker seemed to ignore this for some reason...

**Comment by pv** [comment]
@endolith: please also run `generate_ufuncs.py` under `scipy/special` to apply the changes to `add_newdocs.py`

**Comment by endolith** [comment]
```
Failed example:

print(kn([4, 5, 6], 1))

Expected:

[ 44.23241425 360.96060181 3653.83837891]

Got:

[ 44.23241585 360.9605896 3653.83831186]
```

**Comment by endolith** [comment]
```
>>> print(np.array_str(kn([4, 5, 6], 1), precision=4))
[   44.2324   360.9606  3653.8384]
```

works, but the length distracts from the example

would be better if there were something that printed arrays with reasonable precision by default:

```
>>> np.print(kn([4, 5, 6], 1))
[   44.2324   360.9606  3653.8384]
```

of if the doc tester was smarter about numerical precision

**Comment by endolith** [comment]
Incidentally, is there a reason `kn` exists if `kv` can handle all the same inputs?

**Comment by pv** [comment]
Backward compatibility, mostly.

**Comment by pv** [comment]
@endolith: the checker accepts 1% error, so it should be happy with three decimals.
IOW, `np.set_printoptions(precision=3)` should be fine.

**Comment by pv** [comment]
Ah OK, not sure if the checker understands the array print syntax.
Maybe replacing `print(foo)` by just `foo` is OK.

**Comment by ev-br** [comment]
What Pauli said + make the whole docstring raw to shield the TeX markup ($\nu$). https://github.com/ev-br/scipy/commit/bca778d027c54a06024bb288f5c9436df1f1a5e7 passes refguide-check for me locally.

The array-print output confuses the checker because of the whitespace as a separator. In principle we could special-case a sequence of entries in square brackets separated by whitespace, but I'm not sure it's worth it.

**Comment by endolith** [comment]
Well the only reason I was printing it was to avoid doctest failures if the last few digits varied  :)  http://stackoverflow.com/a/2428921/125507

Argh, I thought I could squash commits and force push and it would reopen the PR

**Pull request closed without merging**